functions/common/geode-common/src/main/java/org/springframework/cloud/fn/common/geode/GeodeClientCacheConfiguration.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.common.geode;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Properties;
+
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.pdx.ReflectionBasedAutoSerializer;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.io.Resource;
+import org.springframework.data.gemfire.util.PropertiesBuilder;
+import org.springframework.util.FileCopyUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author David Turanski
+ * @author Christian Tzolov
+ */
+@EnableConfigurationProperties({ GeodeClientCacheProperties.class, GeodeSecurityProperties.class,
+		GeodeSslProperties.class, GeodePoolProperties.class })
+@Import(InetSocketAddressConverterConfiguration.class)
+public class GeodeClientCacheConfiguration {
+
+	private static final String SECURITY_CLIENT = "security-client-auth-init";
+
+	private static final String SECURITY_USERNAME = "security-username";
+
+	private static final String SECURITY_PASSWORD = "security-password";
+
+	@Bean
+	public ClientCache clientCache(GeodeClientCacheProperties clientCacheProperties,
+			GeodeSecurityProperties securityProperties, GeodeSslProperties sslProperties,
+			GeodePoolProperties geodePoolProperties) {
+
+		Properties properties = new Properties();
```

Comment:
Why do we need this extra object?
Can't we just use that `PropertiesBuilder.setProperty()` API instead?

Reply:
No because it is not always true.

Reply:
It's required to be true for the supplier but not consumer.